other/perun-propagate/etc/perun_propagate

```diff
@@ -39,41 +52,116 @@ check() {
 }
 
 perun_propagate_start() {
+	HOST_NOT_FOUND="doesn't have assigned any facility";
 	CC_FILE=`mktemp`
 	trap "unlink ${CC_FILE}" ERR
 	export KRB5CCNAME="FILE:${CC_FILE}"
 
+	#choose the type of kerberos and prepare Principals (all unique hosts)
 	set -e
 	if ktutil --version </dev/null 2>&1 | grep -qi heimdal; then
 		# Heimdal
-		PRINCIPAL=`ktutil list 2>/dev/null | awk '($3~"^host/") && ($3!~"\\\.ipv6\\\.") { print $3 }' | head -1`
-		kinit --no-afslog --keytab="${KEYTAB}" "${PRINCIPAL}"
+		PRINCIPALS=`ktutil list 2>/dev/null | awk '($3~"^host/") && ($3!~"\\\.ipv6\\\.") { print $3 }' 2>/dev/null | sort | uniq`
+		KERBEROS_TYPE="heimdal"
 	else
 		# MIT Kerberos
-		PRINCIPAL=`echo -e "read_kt ${KEYTAB}\nlist" | ktutil list 2>/dev/null | awk '($3~"^host/") && ($3!~"\\\.ipv6\\\.") { print $3 }' | head -1`
-		kinit -k -t "${KEYTAB}" "${PRINCIPAL}"
+		PRINCIPALS=`echo -e "read_kt ${KEYTAB}\nlist" | ktutil list 2>/dev/null | awk '($3~"^host/") && ($3!~"\\\.ipv6\\\.") { print $3 }' 2>/dev/null | sort | uniq `
+		KERBEROS_TYPE="mit"
 	fi
 
+	#For every unique Principal do kinit and start perun-propagate
 	set +e
-	if which timeout stdbuf &>/dev/null; then
-		OUT=`timeout ${TIMEOUT} stdbuf -oL -eL remctl "${PERUN_SERVER}" perun propagate 2>&1`
-		EXIT_CODE=$?
-	else
-		OUT=`remctl "${PERUN_SERVER}" perun propagate 2>&1`
-		EXIT_CODE=$?
+	ITERATOR=0;
+	while read -r LINE; do
+		if [ "${KERBEROS_TYPE}" = "heimdal" ]; then
+			kinit -k -t "${KEYTAB}" "${LINE}"
+		else	
+			kinit --no-afslog --keytab="${KEYTAB}" "${LINE}"
```

Comment:
Each kinit have to use separate file to store ticket (KRB5CCNAME). Otherwise is possible that the ticket will be overridden before remctl (spawned as separate process) manage to read it.